tools/nymvisor/src/upgrades/mod.rs

```diff
@@ -58,15 +57,17 @@ pub(crate) async fn perform_upgrade(config: &Config) -> Result<UpgradeResult, Ny
             path: lock_path.clone(),
             source,
         })?;
-    let lock_fd = lock_file.as_raw_fd();
 
     debug!("attempting to acquire the lock");
-    if let Err(err) = flock(lock_fd, FlockArg::LockExclusiveNonblock) {
-        return Err(NymvisorError::UnableToAcquireUpgradePlanLock {
-            lock_path,
-            libc_code: err,
-        });
-    }
+    let _lock = match Flock::lock(lock_file, FlockArg::LockExclusiveNonblock) {
```

Comment:
due to the changes to flock, I'd feel better if we tested it before merging. because I'm not sure if `Flock::drop()` will work since it will attempt to unlock the lock. However, by the time it goes out of scope, the file would have been manually removed

Reply:
Yes always good to be cautions, let's park this for now until we have chance to verify it actually working :)